experimental/bot/logger/telemetrylogger/telemetry_logger.go

```diff
@@ -0,0 +1,74 @@
+package telemetrylogger
+
+import (
+	"fmt"
+
+	"github.com/mattermost/mattermost-plugin-api/experimental/bot/logger"
+	"github.com/mattermost/mattermost-plugin-api/experimental/telemetry"
+)
+
+type telemetryLogger struct {
+	logger.Logger
+	logLevel logger.LogLevel
+	tracker  telemetry.Tracker
+}
+
+/*
+New promotes the provided logger into a telemetry logger, storing all events below the logLevel
+through the tracker.
+
+- l Logger: A logger to promote.
+
+- logLevel: The highest type of message to be stored in telemetry.
+
+- tracker: The telemetry tracker to store the messages.
+*/
+func New(l logger.Logger, logLevel logger.LogLevel, tracker telemetry.Tracker) logger.Logger {
```

Comment:
What is the value of passing a `Logger` here instead of a `common.LogAPI`? Why do I have to create a `Logger` first?

Reply:
This way, if we want to create a telemetry logger that also send copies to the admins, we can compose like layers. Something like this:
```
l := admincclogger.New(telemetrylogger.New(logger.New(...), ...), ...)
```
We could also compose something more complex like:
```
l := admincclogger.New(admincclogger.New(logger.New(...), LogLevelError, sysadmins), LogLevelInfo, pluginAdmins)
```

Reply:
We can add another function NewFromAPI to simplify this, but not sure if needed.